x-pack/metricbeat/module/azure/billing/billing.go

```diff
@@ -38,31 +38,38 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	var config azure.Config
 	err := base.Module().UnpackConfig(&config)
 	if err != nil {
-		return nil, errors.Wrap(err, "error unpack raw module config using UnpackConfig")
+		return nil, fmt.Errorf("error unpack raw module config using UnpackConfig: %w", err)
 	}
 	if err != nil {
 		return nil, err
 	}
 	// instantiate monitor client
 	billingClient, err := NewClient(config)
 	if err != nil {
-		return nil, errors.Wrap(err, "error initializing the billing client: module azure - billing metricset")
+		return nil, fmt.Errorf("error initializing the billing client: module azure - billing metricset: %w", err)
 	}
 	return &MetricSet{
 		BaseMetricSet: base,
 		client:        billingClient,
+		log:           logp.NewLogger("azure_billing"),
 	}, nil
 }
 
 // Fetch methods implements the data gathering and data conversion to the right metricset
 // It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
 func (m *MetricSet) Fetch(report mb.ReporterV2) error {
-	results, err := m.client.GetMetrics()
+	startTime := time.Now().UTC().Truncate(24 * time.Hour).Add((-48) * time.Hour)
```

Comment:
OMG, thank you for raising this issue!

The `-48` is a mistake; it should be `-24`. I am not happy to admit it might be a test that ended up in a commit by mistake. I double-checked with the [original version](https://github.com/elastic/beats/blob/c6900091e540bf99b7a1c54637536fa29419613b/x-pack/metricbeat/module/azure/billing/client.go#L58-L59), and now it's correct. I can't believe I made such an error.

Reply:
no problem - glad we caught it!

regarding tests for this sort of thing ... i imagine the test expectation uses similar logic to compute the window. here is a blog post (from a great series from google called 'testing on the toilet') on why this can lead to issues like this one https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

Reply:
i wonder if we could test this better by moving the date window calculation into a separately testable method where we can validate the range is correct?

Reply:
I should have done it from the start.